src/main/java/core/basesyntax/dao/dao/impl/StorageDaoImpl.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.dao.dao.impl;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitModel;
+import java.util.Map;
+
+public class StorageDaoImpl implements StorageDao {
+
+    @Override
+    public String addFruit(FruitModel fruitModel, Integer amount) {
+        for (Map.Entry<FruitModel, Integer> entry: Storage.fruitStorage.entrySet()) {
```

Comment:
you negate all the advantage of hashmap - O(1) constant lookup. Iterating is O(N) - linear. Just retrieve by key